### PR TITLE
seed a Random instance instead of the module

### PR DESCRIPTION
**Pull request by amygdalama**
Seeding the random module will seed that module globally --
this affects calls to random outside the faker package.
Seeding an instance of the Random class will not affect
other calls to the random module.
This addresses one concern in issue #14.

Let me know if you want me to change anything!

### CODE REVIEW

File: faker/tests/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+mock==1.0.1
```
Comment (amygdalama) [inline faker/tests/requirements.txt]: I used `mock` for the test I added, but I'm not sure how y'all would like to specify this requirement. It only needs to be installed for python < 3.3.

Adding a requirement here means in order to run tests, you'd need to run
```
$ pip install -r tests/requirements.txt
```
This is how django handles this problem: https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/unit-tests/#running-unit-tests-dependencies

Is there another way you'd like to handle this? If we stick with this route, I should probably add the requirement to the documentation somewhere, but I wanted to run this by y'all first to see if this is how you'd like to handle it.

Comment (fcurella) [inline faker/tests/requirements.txt]: I think `tests/requirements.txt` is the way to go.

The various `travis.yml` and co. will need to be updated to run `pip install -r tests/requirements.txt`

Comment (amygdalama) [inline faker/tests/requirements.txt]: Great. I updated the `*.yml`s (I needed to add one for CircleCI) and updated the documentation in the README for running tests.

Any other changes you'd like me to make while waiting for tests to finish?

Comment (fcurella) [inline faker/tests/requirements.txt]: Nope, :sparkles:looks perfect :sparkles:!

Thank you so much!

Comment (amygdalama) [inline faker/tests/requirements.txt]: You're welcome! Could you cut a release with this change? We're waiting on it to merge some changes for our test suite.

Comment (fcurella) [inline faker/tests/requirements.txt]: Releases will have to be coordinated with the original author. I can bump up version numbers but I can't upload to PyPI. Would that work for you?

I could ping the author about uploaidng a release, but he's currently busy working on his Master's Thesis and I'd like let him focus on that :)

Comment (amygdalama) [inline faker/tests/requirements.txt]: Bumping the version works too! Thanks :smile:

Comment (fcurella) [inline faker/tests/requirements.txt]: Done :)

**Comment by amygdalama** [comment]
FYI I released a nose plugin that will seed faker based on the test name before each test: https://github.com/venmo/nose-seed-faker :smile:

**Comment by fcurella** [comment]
Thanks @amygdalama ! That's a great idea!

@joke2k can we upload a release to PyPI?

**Pull request merged**